awscli/customizations/cloudfront.py

```diff
@@ -49,3 +55,64 @@ def add_to_params(self, parameters, value):
                 "CallerReference": caller_reference,
                 "Paths": {"Quantity": len(value), "Items": value},
                 }
+
+
+def _add_sign(command_table, session, **kwargs):
+    command_table['sign'] = SignCommand(session)
+
+
+class SignCommand(BasicCommand):
+    NAME = 'sign'
+    DESCRIPTION = 'Sign a given url.'
+    DATE_HELP = """Supported formats include:
+        YYYY-MM-DD (which means 0AM UTC of that day),
+        YYYY-MM-DDThh:mm:ss (with default timezone as UTC),
+        YYYY-MM-DDThh:mm:ss+hh:mm or YYYY-MM-DDThh:mm:ss-hh:mm (with offset),
+        or EpochTime (which always means UTC).
+        Do NOT use YYYYMMDD, because it will be treated as EpochTime."""
+    ARG_TABLE = [
+        {
+            'name': 'url',
+            'no_paramfile': True,  # To disable the default paramfile behavior
+            'required': True,
+            'help_text': 'The URL to be signed',
+        },
+        {
+            'name': 'key-pair-id',
+            'required': True,
+            'help_text': 'The ID of your CloudFront key pairs.',
+        },
+        {
+            'name': 'private-key',
+            'required': True,
+            'help_text': 'file://path/to/your/private-key.pem',
+        },
+        {'name': 'date-less-than', 'required': True, 'help_text': DATE_HELP},
+        {'name': 'date-greater-than', 'help_text': DATE_HELP},
+        {'name': 'ip-address', 'help_text': 'Format: x.x.x.x/x or x.x.x.x'},
+    ]
+
+    def _run_main(self, args, parsed_globals):
+        signer = CloudFrontSigner(
+            args.key_pair_id, RSASigner(args.private_key).sign)
+        date_less_than = parse_to_aware_datetime(args.date_less_than)
+        if args.date_greater_than is not None:
```

Comment:
``args.date_greater_than`` is set to ``None`` if not provided so you can simplify the logic:
```py
date_greater_than = args.date_greater_than
if date_greater_than is None:
     date_greater_than = parse_to_aware_datetime(date_greater_than)
```